extension/bootstrap.js

```diff
@@ -163,43 +173,48 @@ class CopyController {
 
   supportsCommand(cmd) { return cmd === "cmd_copy" || cmd === "share-button-study"; }
 
-  isCommandEnabled(cmd) { return true; }
+  isCommandEnabled() { return true; }
 
-  doCommand(cmd) {
+  async doCommand(cmd) {
     if (cmd === "cmd_copy") {
-      studyUtils.telemetry({ event: "copy" });
+      // Iterate over all other controllers and call doCommand on the first controller
+      // that supports it
+      // Skip until we reach the controller that we inserted
+      let i = 0;
+      const urlInput = this.browserWindow.urlInput;
+
+      for (; i < urlInput.controllers.getControllerCount(); i++) {
+        const curController = urlInput.controllers.getControllerAt(i);
+        if (curController.supportsCommand("share-button-study")) {
+          i += 1;
+          break;
+        }
+      }
+      for (; i < urlInput.controllers.getControllerCount(); i++) {
+        const curController = urlInput.controllers.getControllerAt(i);
+        if (curController.supportsCommand(cmd)) {
+          curController.doCommand(cmd);
+          break;
+        }
+      }
+
+      const pingData = { event: "copy" };
+      await studyUtils.telemetry(pingData);
+      // FIXME Cannot await within CopyController
+      await PingStorage.logPing(pingData);
       const shareButton = this.browserWindow.shareButton;
       // check to see if we should call a treatment at all
       const numberOfTimeShown = Preferences.get(COUNTER_PREF, 0);
       if (numberOfTimeShown < MAX_TIMES_TO_SHOW &&
           Object.keys(TREATMENTS).includes(this.treatment)) {
         Preferences.set(COUNTER_PREF, numberOfTimeShown + 1);
-        TREATMENTS[this.treatment](this.browserWindow, shareButton);
-      }
-    }
-    // Iterate over all other controllers and call doCommand on the first controller
-    // that supports it
-    // Skip until we reach the controller that we inserted
-    let i = 0;
-    const urlInput = this.browserWindow.urlInput;
-
-    for (; i < urlInput.controllers.getControllerCount(); i++) {
-      const curController = urlInput.controllers.getControllerAt(i);
-      if (curController.supportsCommand("share-button-study")) {
-        i += 1;
-        break;
-      }
-    }
-    for (; i < urlInput.controllers.getControllerCount(); i++) {
-      const curController = urlInput.controllers.getControllerAt(i);
-      if (curController.supportsCommand(cmd)) {
-        curController.doCommand(cmd);
-        break;
+        // FIXME Cannot await within CopyController
+        await TREATMENTS[this.treatment](this.browserWindow, shareButton);
```

Comment:
You say "cannot await", and then immediately await? I'm not sure I understand this comment.